app-class.js

```diff
@@ -0,0 +1,55 @@
+class BookManager {
+  constructor() {
+    this.books = [];
+    this.id = 0;
+    this.form = document.getElementById('form');
+    this.bookSection = document.getElementById('books');
+
+    this.initialize();
+  }
+
+  initialize() {
+    this.form.addEventListener('submit', this.callbackFunction.bind(this));
+    this.displayData();
+  }
+
+  displayData() {
+    const storedBooks = localStorage.getItem('books');
+    this.books = storedBooks ? JSON.parse(storedBooks) : [];
+    this.bookSection.innerHTML = '';
+
+    this.books.forEach((book, index) => {
+      const bookElement = document.createElement('div');
+      bookElement.classList.add('book');
+      bookElement.innerHTML = `
+        <p id="title">"${book.title}" by ${book.author}</p>
+        <button class='btn' onclick='bookManager.removeBook(${index})'>Remove</button>`;
+      this.bookSection.appendChild(bookElement);
+    });
+  }
+
+  callbackFunction(event) {
+    event.preventDefault();
+    const myFormData = new FormData(event.target);
+    let formDataObj = {};
+    myFormData.forEach((value, key) => {
+      formDataObj[key] = value;
+    });
+    this.id += 1;
+    formDataObj = { ...formDataObj, id: this.id };
+    this.books.push(formDataObj);
+    const storedBook = JSON.stringify(this.books);
+    localStorage.setItem('books', storedBook);
+    this.displayData();
+  }
+
+  removeBook(index) {
+    this.books.splice(index, 1);
```

Comment:
- [ ] [OPTIONAL]
Again, it will be a good idea if we use a non-mutating array method such as `slice` over `splice`. In this way, we avoid modifying the original array.